appinventor/components/src/com/google/appinventor/components/runtime/ListView.java

```diff
@@ -343,7 +426,39 @@ public String Selection(){
   public void Selection(String value) {
     selection = value;
     // Now, we need to change SelectionIndex to correspond to Selection.
-    selectionIndex = ElementsUtil.setSelectedIndexFromValue(value, items);
+    if(!currentItems.isEmpty()) {
+      selectionIndex = ElementsUtil.setSelectedIndexFromValue(value, currentItems);
+      selectionDetailText = ElementsUtil.setDetailSelectionFromIndex(selectionIndex, currentItems.get(selectionIndex-1));
+    } else {
+      selectionIndex = ElementsUtil.setSelectedIndexFromValue(value, items);
+      selectionDetailText = "";
+    }
+  }
+
+  /**
+   * Returns the Secondary or Detail text in the ListView at the position set by SelectionIndex
+   */
+  @SimpleProperty(description = "Returns the secondary text of the selected row in the ListView.",
+      category = PropertyCategory.BEHAVIOR)
+  public String SelectionDetailText(){
+    return selectionDetailText;
+  }
+
+  /**
+   * SelectionDetailText property setter method
+   */
+  @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_STRING,
```

Comment:
I added this thinking just in case the user wants to do something with this, maybe with getter. Should I remove this property entirely? Or only from the designer?